docs/setup/configuration.md
# Configuration

## Autolock account

For security reason, a mechanism locking automatically unused account
has been developed.

When an account is created and the user never logged in, the account is
automatically disabled after a certain amount of time and a mail is sent
to the user.

When an account is not used during a customizable amount of time, it's
automatically blocked and a mail is also sent to the user.

It can be customized using the following parameters in the settings.py
file:

```python
# After AUTOLOCK_DELAY days of inactivity account will be blocked
AUTOLOCK_DELAY = 180

# Mail sent when an account is auto locked
AUTOLOCK_TEMPLATE = u"""Hello %s

You have an active account on Swordphish but you never logged in.

As a security measure, your account has been locked. If you need it reply this mail!

Best regards,

Swordphish administrators
"""

# AUTOLOCK_NEVER_USED_DELAY days after creation an account will be locked if not used
AUTOLOCK_NEVER_USED_DELAY = 30

# Mail sent when an account is never used
AUTOLOCK_NEVER_USED_TEMPLATE = u"""Hello %s

You have an active account on Swordphish but didn't use during the last %s days.

As a security measure, your account has been locked. If you need it reply this mail!

Best regards,

Swordphish administrators
"""
```

## Autoclean

To prevent the accumulation of campaigns results and targets lists in
time, we've developed an autoclean feature which automatically deletes
the lists and the campaigns results after a customizable amount of time.

It can be customized using the following parameters in the settings.py
file:

```python
# After AUTOCLEAN_DELAY days the campaigns / targets will be automatically deleted
AUTOCLEAN_DELAY = 90

# The day of the week when the auto delete is performed
AUTOCLEAN_DAY = "saturday"
```

## Special mail header

A special header is automatically added in the phishing mails sent by
swordphish.

The name of this header can be customized in settings.py file, by
default the value is:

```python
PHISHING_MAIL_HEADER = "X-Swordphish-Awareness-Campaign"
```

The header contains the target-id between brackets, ex:

```
    X-Sworphish-Awareness-Campaign: [0916e333-ad79-4102-a220-13dbd7ad4195]
```

You can use this to extract automatically the ID and count the mail as
reported using a plugin in your mail client using the report API (see
below).

## Report API

The report API allows to mark the mail as "reported" programmatically.
It's used at Société Générale in our Outlook plugin. When the mail is
detected as swordphish campaign by the plugin using the customized
header, the plugin then sends an HTTP GET request to this URL to count
the mail as reported.

The URL is structured like this:

```
    https?://swordphish_hostname/result/report/<target-id>
```

Example:

```
    https?://swordphish_hostname/result/report/0916e333-ad79-4102-a220-13dbd7ad4195
```

## Phishing domain feed

A json feed listing all the phishing domains is available on the
following URL:

```
    https?://swordphish_hosntame/2985836c7501af76a0bdd92f1d120cd2/domains_feed
```

This feature can be useful to integrate this feed in your security
monitoring tools automatically (SIEM) so your security teams (SOC /
CERT) will know if this suspicious domain is linked to a phishing
awareness campaign.


Swordphish/settings.py
"""
Django settings for SwordPhish project.

For more information on this file, see
https://docs.djangoproject.com/en/1.7/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/1.7/ref/settings/
"""

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
import os

from django.conf import settings

BASE_DIR = os.path.dirname(os.path.dirname(__file__))

# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/1.7/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!

SECRET_KEY = 'REPLACE_IT_IN_PRODUCTION'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = ['localhost', "127.0.0.1", "::1"]
CSRF_TRUSTED_ORIGINS = ['http://' + h for h in ALLOWED_HOSTS] + ['https://' + h for h in ALLOWED_HOSTS]

PROJECT_NAME = "SwordPhish"

# Application definition

INSTALLED_APPS = (
    'django.contrib.auth',
    'django.contrib.admin',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django_bootstrap5',
    'tempus_dominus',
    'ckeditor',
    'django_select2',
    'LocalUsers',
    'Main',
)

MIDDLEWARE = (
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'django.middleware.locale.LocaleMiddleware',
)

ROOT_URLCONF = 'Swordphish.urls'

WSGI_APPLICATION = 'Swordphish.wsgi.application'

# Database
# https://docs.djangoproject.com/en/1.7/ref/settings/#databases

DEFAULT_AUTO_FIELD = 'django.db.models.AutoField'

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': 'db.sqlite3',
    }
}

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.contrib.auth.context_processors.auth',
                'django.template.context_processors.debug',
                'django.template.context_processors.i18n',
                'django.template.context_processors.media',
                'django.template.context_processors.static',
                'django.template.context_processors.tz',
                'django.template.context_processors.request',
                'django.contrib.messages.context_processors.messages',
            ],
            'debug': True,
        },
    },
]

# Internationalization
# https://docs.djangoproject.com/en/1.7/topics/i18n/

LANGUAGE_CODE = 'en-us'

LANGUAGES = [
    ('en', 'English'),
]

TIME_ZONE = 'Europe/Paris'

USE_I18N = True

USE_L10N = True

USE_TZ = True

LOCALE_PATHS = ('locale',)

# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/1.7/howto/static-files/
STATIC_ROOT = os.path.join(BASE_DIR, 'static')
# MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
STATIC_URL = '/static/'
MEDIA_URL = '/media/'

LOGIN_URL = 'Authent:login'
LOGOUT_URL = 'Authent:logout'

SESSION_SAVE_EVERY_REQUEST = True

# Default settings
BOOTSTRAP5 = {
    "css_url": {
        "url": "https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css",
        "integrity": "sha384-QWTKZyjpPEjISv5WaRU9OFeRpok6YctnYmDr5pNlyT2bRjXh0JMhjY6hW+ALEwIH",
        "crossorigin": "anonymous",
    },
    "javascript_url": {
        "url": "https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js",
        "integrity": "sha384-YvpcrYf0tY3lHB60NNkmXc5s9fDVZLESaAA55NDzOxhy9GkcIdslK1eN7N6jIeHz",
        "crossorigin": "anonymous",
    },
    # Class to indicate error (better to set this in your Django form)
    'error_css_class': 'has-error',

    # Color mode: light/dark (None means do not set color mode).
    "color_mode": None,
}

CKEDITOR_BASEPATH = STATIC_URL + "ckeditor/ckeditor"

CKEDITOR_CONFIGS = {
    'default': {
        'toolbar': 'Custom',
        'toolbar_Custom': [
            ['Bold', 'Italic', 'Underline'],
            ['Table', 'NumberedList', 'BulletedList', 'HorizontalRule'],
            ['Outdent', 'Indent'],
            ['JustifyLeft', 'JustifyCenter', 'JustifyRight', 'JustifyBlock'],
            ['TextColor', 'BGColor'], ['base64image', 'Link', 'Unlink'],
            ['Styles', 'Format', 'Font', 'FontSize'],
            ['RemoveFormat', 'Source', 'DocProps', 'Maximize']
        ],
        'extraPlugins': 'base64image,docprops,font,maximize',
        'width': '100%',
        'fullPage': 'true',
        'removeDialogTabs': 'link:Upload;link:upload;link:advanced;docProps:meta;docProps:preview;',
        'filebrowserBrowseUrl': '',
        'filebrowserImageBrowseUrl': '',
        'filebrowserFlashBrowseUrl': '',
    },
    'withforms': {
        'toolbar': 'Custom',
        'toolbar_Custom': [
            ['Bold', 'Italic', 'Underline'],
            ['Table', 'NumberedList', 'BulletedList', 'HorizontalRule'],
            ['Outdent', 'Indent'],
            ['JustifyLeft', 'JustifyCenter', 'JustifyRight', 'JustifyBlock'],
            ['TextColor', 'BGColor'], ['base64image', 'Link', 'Unlink'],
            ['Styles', 'Format', 'Font', 'FontSize'],
            ['RemoveFormat', 'Source', 'DocProps', 'Maximize'],
            ['Form', 'TextField', 'Button']
        ],
        'extraPlugins': 'base64image,docprops,font,maximize,forms',
        'width': '100%',
        'fullPage': 'true',
        'removeDialogTabs': 'link:Upload;link:upload;link:advanced;docProps:meta;docProps:preview;',
        'filebrowserBrowseUrl': '',
        'filebrowserImageBrowseUrl': '',
        'filebrowserFlashBrowseUrl': '',
    }
}

# Mail address used to send mails
SERVER_EMAIL = "SwordPhish <swordphish@invalid.notld>"

# Mail server information
EMAIL_HOST = "localhost"
EMAIL_PORT = 2525
# EMAIL_HOST_USER = "FIXME"
# EMAIL_HOST_PASSWORD = "FIXME"

# Used to alert when a campaign starts
ALERT_RECIPIENTS = ["fixme@invalid.notld"]

# After AUTOLOCK_DELAY days of inactivity account will be blocked
AUTOLOCK_DELAY = 180

# Mail sent when an account is auto locked
AUTOLOCK_TEMPLATE = u"""Hello %s

You have an active SwordPhish account but you never logged in.

As a security measure, your account has been locked. If you need it reply to this email!

Best regards,
SwordPhish team
"""

# AUTOLOCK_NEVER_USED_DELAY days after creation an account will be locked if not used
AUTOLOCK_NEVER_USED_DELAY = 30

# Mail sent when an account is never used
AUTOLOCK_NEVER_USED_TEMPLATE = u"""Hello %s

You have an active SwordPhish account but haven't used it during the last %s days.

As a security measure, your account has been locked. If you need it reply to this email!

Best regards,
SwordPhish team
"""

# After AUTOCLEAN_DELAY days the campaigns / targets will be automatically deleted
AUTOCLEAN_DELAY = 90

# The day of the week when the auto delete is performed
AUTOCLEAN_DAY = "saturday"

# Phishing mail header
PHISHING_MAIL_HEADER = "X-Swordphish-Awareness-Campaign"

LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'handlers': {
        'console': {
            'class': 'logging.StreamHandler',
        },
    },
    'loggers': {
        'django': {
            'handlers': ['console'],
            'level': os.getenv('DJANGO_LOG_LEVEL', 'INFO'),
        },
    },
}
